### PR TITLE
allow to switch content language on list and filter by language

### LINKED ISSUE #81: Translation on list
Hello,

You recently added the translation for the object show. 

I need the same on the object list. 

I also need the filters to use the translation on list, filter on every locale available...

Thanks.

### PR DESCRIPTION
**Pull request by dbu**
fix #81, fix #83

continuation of #83, #84

### CODE REVIEW

**Comment by dbu** [comment]
@mcrinquand can you please review this PR and check if it still works as intended? and it would be awesome if you could add a bit of documentation how this is to be used.

File: Resources/views/Block/block_locale_switcher.html.twig [hunk 1]
```diff
@@ -2,8 +2,15 @@
 {% set admin  = block_context.settings.admin %}
 {% set locale_switcher_route = block_context.settings.locale_switcher_route %}
 {% set locale_switcher_route_parameters = block_context.settings.locale_switcher_route_parameters %}
+{% set currentLocale = object.locale %}
```
Comment (skyosev) [inline Resources/views/Block/block_locale_switcher.html.twig]: `object` is null in the list route, so trying to access its property throws an exception.

File: Admin/Extension/Gedmo/TranslatableAdminExtension.php [hunk 2]
```diff
@@ -40,14 +42,71 @@ public function alterObject(AdminInterface $admin, $object)
...
+     * doctrine_orm_callback filter type.
+     *
+     * @param ProxyQuery $queryBuilder
+     * @param string     $alias
+     * @param string     $field
+     * @param string     $value
+     *
+     * @return bool|null
+     */
+    public static function translationFieldFilter(ProxyQuery $queryBuilder, $alias, $field, $value)
```
Comment (skyosev) [inline Admin/Extension/Gedmo/TranslatableAdminExtension.php]: This is never used or at least no hint how to use it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## 2.1.0 (unreleased)
```
Comment (core23) [inline CHANGELOG.md]: You can remove this line. We add this line when releasing a new version.

**Comment by dbu** [comment]
@mcrinquand seems to not respond. if somebody has the time to wrap this up, would be cool. i only attempted to clean up the code, but don't have much time to follow up.

**Comment by OskarStark** [comment]
any news on this PR @dbu ?

**Comment by dbu** [comment]
no updates. i only updated the code started by @mcrinquand but don't have time to wrap this up.

**Comment by dmarkowicz** [comment]
I merged conflicts, fixed null exception and add documentation for `translationFieldFilter` here: https://github.com/dmarkowicz/SonataTranslationBundle/compare/2.x...translate-gedmo-lists. Should I open another PR or is there any other solution to continue this one?

**Comment by greg0ire** [comment]
I think you should open a new one.

**Comment by dmarkowicz** [comment]
It may be closed because #143 already merged.

**Pull request closed without merging**